### PR TITLE
fix: propagate try_make_sync_target_canonical error

### PR DESCRIPTION
**Pull request by Rjected**
This fixes an issue where an invalid downloaded block could be reported as VALID if it is the current sync target, since checks for state root would occur in `try_make_sync_target_canonical`, which was infallible.

### CODE REVIEW

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -1484,7 +1518,10 @@ where
     /// Note: This will not succeed if the sync target has changed since the block download request
     /// was issued and the new target is still disconnected and additional missing blocks are
     /// downloaded
-    fn try_make_sync_target_canonical(&mut self, inserted: BlockNumHash) {
+    fn try_make_sync_target_canonical(
+        &mut self,
+        inserted: BlockNumHash,
+    ) -> Result<(), (B256, RethError)> {
```
Comment (rkrasiuk) [inline crates/consensus/beacon/src/engine/mod.rs:1524]: we need to clean this up in the future, make better typed responses

File: crates/consensus/beacon/src/engine/mod.rs [hunk 2]
```diff
@@ -1120,7 +1120,30 @@ where
...
+                                    // If we could not make the sync target block canonical, we
+                                    // should return the error as an invalid payload status.
+                                    return Ok(PayloadStatus::new(
+                                        PayloadStatusEnum::Invalid {
+                                            validation_error: error.to_string(),
+                                        },
+                                        // TODO: return a proper latest valid hash
+                                        //
+                                        // See: <https://github.com/paradigmxyz/reth/issues/7146>
+                                        self.forkchoice_state_tracker.last_valid_head(),
```
Comment (rkrasiuk) [inline crates/consensus/beacon/src/engine/mod.rs:1143]: kk

**Review by rkrasiuk — APPROVED** [review]
lgtm

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**